src/llvm/mod.rs

```diff
@@ -65,18 +64,8 @@ pub(crate) fn init(args: &[Cow<'_, CStr>], overview: &CStr) {
     unsafe { LLVMParseCommandLineOptions(c_ptrs.len() as i32, c_ptrs.as_ptr(), overview.as_ptr()) };
 }
 
-pub(crate) fn create_module(name: &CStr, context: LLVMContextRef) -> Option<LLVMModuleRef> {
```

Comment:
here is an example of an API I had changed (to accept CStr) to avoid an unwrap, which is lost in this rebase.

Reply:
yeah, it was very difficult to rebase, I had to constantly look at your changes. I have to double check everything, this was my first round. 

And I would like to also find a safer solution for the diagnostic handler. Pin locks the memory address but doesn't ensure it remains alive